src/Symfony/Component/VarDumper/Command/ServerDumpCommand.php

```diff
@@ -0,0 +1,301 @@
+ */
+class ServerDumpCommand extends Command
+{
+    private static $style = <<<'CSS'
+body {
+    display: flex;
+    flex-direction: column-reverse;
+    justify-content: flex-end;
+    word-wrap: break-word;
+    background-color: #F9F9F9;
+    color: #222;
+    font-family: Helvetica, Arial, sans-serif;
+    font-size: 14px;
+    line-height: 1.4;
+}
+p {
+    margin: 0;
+}
+a {
+    color: #218BC3;
+    text-decoration: none;
+}
+a:hover {
+    text-decoration: underline;
+}
+code {
+    color: #cc2255;
+    background-color: #f7f7f9;
+    border: 1px solid #e1e1e8;
+    border-radius: 3px;
+    margin-right: 5px;
+    padding: 0 3px;
+}
+.text-small {
+    font-size: 12px !important;
+}
+article {
+    margin: 5px;
+    margin-bottom: 10px;
+}
+article > header {
+    display: flex;
+    flex-direction: row;
+    align-items: baseline;
+}
+article > header > * {
+    flex: 1;
+    display: flex;
+    align-items: baseline;
+}
+article > header > h2 {
+    font-size: 14px;
+    color: #222;
+    font-weight: normal;
+    font-family: "Lucida Console", monospace, sans-serif;
+    word-break: break-all;
+    margin-right: 5px;
+    user-select: all;
+}
+article > header > h2 > code {
+    white-space: nowrap;
+    user-select: none;
+}
+article > header > span.date {
+    flex: 0;
+    text-align: right;
+    white-space: nowrap;
+    color: #999;
+    font-style: italic;
+}
+article > section.body {
+    border: 1px solid #d8d8d8;
+    background: #FFF;
+    padding: 10px;
+    border-radius: 3px;
+}
+pre.sf-dump {
+    border-radius: 3px;
+    margin-bottom: 0;
+}
+.hidden {
+    display: none; !important
+}
+CSS;
+
+    private static $javascript = <<<'JS'
+document.addEventListener('DOMContentLoaded', function() {
+    let prev = null;
+    Array.from(document.getElementsByTagName('article')).reverse().forEach(function (article) {
+      const dedupId = article.dataset.dedupId;
+      if (dedupId === prev) {
+        article.getElementsByTagName('header')[0].classList.add('hidden');
+      }
+      prev = dedupId;
+    });
+});
+JS;
+
+    private $cliDumper;
+    private $htmlDumper;
+    private $fileLinkFormatter;
+    private $projectDir;
+    private $logger;
+    /** @var Filesystem */
+    private $filesystem;
+    private $lastIdentifier;
+
+    public function __construct(CliDumper $cliDumper = null, HtmlDumper $htmlDumper = null, FileLinkFormatter $fileLinkFormatter, $projectDir = null, LoggerInterface $logger = null)
+    {
+        $this->cliDumper = $cliDumper;
+        $this->htmlDumper = $htmlDumper;
+        $this->fileLinkFormatter = $fileLinkFormatter;
+        $this->projectDir = $projectDir;
+        $this->logger = $logger;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('server:dump')
+            ->addOption('output', null, InputOption::VALUE_REQUIRED, 'An file path to save dumped data in html format, or null to display on CLI output.')
```

Comment:
`A file ...`

Reply:
What about replacing the `--output` option by a `--format` option and extract processing in proper descriptors? It would let the user redirect or pipe the output and may allow other streamable formats like json-text-sequence to be implemented next.
Extracting descriptors from the command might also answer https://github.com/symfony/symfony/pull/23831#discussion_r135015342 by allowing reuse and easing tests.

Reply:
Done.